libraries/AP_GPS/AP_GPS.h

```diff
@@ -297,6 +301,16 @@ class AP_GPS
         return location(primary_instance);
     }
 
+    // Height (meters) above the WGS84 ellipsoid
+    float height_above_WGS84(uint8_t instance) const {
```

Comment:
I'd rather this be a pass by reference on the height, and use a boolean return to flag that we did or did not have the height. It just makes for an externally more consistent API.

Reply:
Agree. I was following the existing pattern though?

Reply:
I'm aware the current pattern was that, I'm trying to slowly factor away from that though, as it causes us a lot of problems with forgetting to check somewhere. It also means we can usually remove the `have_*` functions if it's cheap enough, because just getting the value is cheap, and we were probably about to do it anyways.